javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -134,8 +137,20 @@ final class Pattern private[regex] (
 
   private[regex] def getIndices(lastMatch: js.RegExp.ExecResult, forMatches: Boolean): IndicesArray = {
     val lastMatchDyn = lastMatch.asInstanceOf[js.Dynamic]
-    if (js.isUndefined(lastMatchDyn.indices))
-      lastMatchDyn.indices = indicesBuilder(forMatches, lastMatch.input, lastMatch.index)
+    if (js.isUndefined(lastMatchDyn.indices)) {
+      if (supportsIndices) {
+        if (!enabledNativeIndices) {
+          jsRegExpForFind = new js.RegExp(jsPattern, jsFlagsForFind + "d")
+          jsRegExpForMatches = new js.RegExp(wrapJSPatternForMatches(jsPattern), jsFlags + "d")
+          enabledNativeIndices = true
+        }
+        val regexp = if (forMatches) jsRegExpForMatches else jsRegExpForFind
+        regexp.lastIndex = lastMatch.index
+        lastMatchDyn.indices = regexp.exec(lastMatch.input).asInstanceOf[js.Dynamic].indices
```

Comment:
Consider avoiding the second execution if `lastMatch.indices` is already defined.

Reply:
Hum, I'm not sure I understand. Isn't that what the following line, above, does:
```scala
if (js.isUndefined(lastMatchDyn.indices)) {
```

Reply:
I don't understand either... Seems fine :P